src/VirtoCommerce.DemoCustomerSegmentsModule.Web/Scripts/blades/customerSegment-list.js

```diff
@@ -0,0 +1,149 @@
+                    take: $scope.pageSettings.itemsPerPageCount
+                };
+            }
+
+            $scope.selectNode = function (node) {
+                $scope.selectedNodeId = node.id;
+
+                var newBlade = {
+                    id: 'customerSegmentDetail',
+                    originalEntity: node,
+                    title: node.name,
+                    subtitle: 'demoCustomerSegmentsModule.blades.customer-segment-detail.subtitle',
+                    controller: 'virtoCommerce.DemoCustomerSegmentsModule.customerSegmentDetailController',
+                    template: 'Modules/$(virtoCommerce.DemoCustomerSegmentsModule)/Scripts/blades/customerSegment-detail.tpl.html'
+                };
+
+                bladeNavigationService.showBlade(newBlade, blade);
+            };
+
+            $scope.deleteSegments = function (list) {
+                var dialog = {
+                    id: "confirmDeleteItem",
+                    title: "demoCustomerSegmentsModule.dialogs.customer-segment-delete.title",
+                    message: "demoCustomerSegmentsModule.dialogs.customer-segment-delete.message",
+                    messageValues: list.length > 1 ? {segments: 'segments', count: 'segments'} : {segments: `segment ${list[0].name}`, count: 'segment'},
+                    callback: function (remove) {
+                        if (remove) {
+                            bladeNavigationService.closeChildrenBlades(blade, function () {
+                                blade.isLoading = true;
+                                var itemIds = _.pluck(list, 'id');
+                                customerSegmentsApi.delete({ ids: itemIds }, function () {
+                                    blade.refresh();
+                                });
+                            });
+                        }
+                    }
+                };
+                dialogService.showConfirmationDialog(dialog);
+            };
+
+            var filter = blade.filter = $scope.filter = {};
+            filter.criteriaChanged = function () {
+                if ($scope.pageSettings.currentPage > 1) {
+                    $scope.pageSettings.currentPage = 1;
+                } else {
+                    blade.refresh();
+                }
+            };
+
+            // ui-grid
+            $scope.setGridOptions = function (gridOptions) {
+                bladeUtils.initializePagination($scope, true);
+                $scope.pageSettings.itemsPerPageCount = 20;
+
+                uiGridHelper.initialize($scope, gridOptions, function (gridApi) {
+                    //update gridApi for current grid
+                    $scope.gridApi = gridApi;
+                    uiGridHelper.bindRefreshOnSortChanged($scope);
+                    $scope.gridApi.infiniteScroll.on.needLoadMoreData($scope, showMore);
+
+                });
+
+                $timeout(function () {  blade.refresh(); });
```

Comment:
What for you using $timeout service here?

Reply:
done, removed from code